src/tests/baseservices/exceptions/unhandled/unhandled.cs

```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace TestUnhandledException
+{
+    class Program
+    {
+        static int Main(string[] args)
+        {
+            if (args.Length != 0)
+            {
+                throw new Exception("Test");
+            }
+
+            List<string> lines = new List<string>();
+
+            Process testProcess = new Process();
+
+            testProcess.StartInfo.FileName = Environment.ProcessPath;
+            testProcess.StartInfo.Arguments = Environment.CommandLine + " throw";
+            testProcess.StartInfo.UseShellExecute = false;
+            testProcess.StartInfo.RedirectStandardError = true;
+            testProcess.ErrorDataReceived += (sender, line) => 
+            {
+                Console.WriteLine($"\"{line.Data}\"");
+                if (!string.IsNullOrEmpty(line.Data))
+                {
+                    lines.Add(line.Data);
+                }
+            };
+
+            testProcess.Start();
+            testProcess.BeginErrorReadLine();
+            testProcess.WaitForExit();
+            testProcess.CancelErrorRead();
+
+            int expectedExitCode;
+            if ((Environment.OSVersion.Platform == PlatformID.Unix) || (Environment.OSVersion.Platform == PlatformID.MacOSX))
```

Comment:
```suggestion
            if (!OperatingSystem.IsWindows())
```
`Environment.OSVersion.Platform` is a soft-obsolete API. It behaves in non-sensical way for historic reasons (for example, it returnns `PlatformID.Unix` even on macOS).